cwms-data-api/src/main/java/cwms/cda/api/TimeSeriesController.java

```diff
@@ -411,6 +411,26 @@ public void getAll(@NotNull Context ctx) {
             String begin = ctx.queryParam(BEGIN);
             String end = ctx.queryParam(END);
             String timezone = ctx.queryParamAsClass(TIMEZONE, String.class).getOrDefault("UTC");
+            String versionDateParam = ctx.queryParam(VERSION_DATE);
+            ZonedDateTime versionDate = null;
+            if (versionDateParam != null) {
+                versionDate = DateUtils.parseUserDate(versionDateParam, timezone);
+            }
+
+            String versionType = queryParamAsClass(ctx,
+                    new String[]{VERSION_TYPE, },
+                    String.class, VersionType.MAX_AGGREGATE.getValue(), metrics,
+                    name(TimeSeriesController.class.getName(), GET_ONE));
+
+            // Throw error if max aggregate version type is requested with a version date
+            // or single version version type is requested without a version date
+            if(Objects.equals(versionType, "MAX_AGGREGATE") && !Objects.equals(versionDate, null)) {
+                throw new IllegalArgumentException("Cannot query a max aggregate with a version date.");
+            }
+            if(Objects.equals(versionType, "SINGLE_VERSION") && Objects.equals(versionDate, null)) {
```

Comment:
Note to self: There is test coverage of these cases?

Reply:
yes